index.js

```diff
@@ -9,10 +9,9 @@ const { Tags, FORMAT_HTTP_HEADERS } = opentracing
 
 function jaegerPlugin (fastify, opts, next) {
   assert(opts.serviceName, 'Jaeger Plugin requires serviceName option')
+  const { state = {}, initTracerOpts = {}, ...tracerConfig } = opts
```

Comment:
Im inclined to just want to have a single object that corresponds to the tracer configuration as a whole, at the end initTracer function exported by jaeger client, will use only the parameters necessary to configure the client, without explicitly making a separation between initialization options for the tracer and configuration options.

That being said, any specific reason why adding this to the API vs the current approach is a better?

Reply:
With the current approach the plugin breaks if you attempt to pass `opts` that contain any `reporter` or `throttler` configuration. Alternatively we could keep the API as is and just remove the ability to configure the the second param of `initTrace`.

Reply:
Since tracerDefaults containted opts and was being spread into both config and options params, any opts containing the reporter or throttler key would cause the plugin to crash ("TypeError: this._reporter.setProcess is not a function")